### PR TITLE
[Collection] Docker updates to #3

### PR DESCRIPTION
**Pull request by tmichelet**
Review fixes:
- reduce code duplication
- Prefix auditor entries
- introduce pipelines to cleanup which channel a input uses

@egueidan Commits should be easy to review separately

see also #3

### CODE REVIEW

File: pkg/pipeline/provider.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+func (pp *PipelineProvider) MockPipelineChans() {
+	pp.pipelinesChans = [](chan message.Message){}
+	pp.pipelinesChans = append(pp.pipelinesChans, make(chan message.Message))
+	pp.numberOfPipelines = 1
+}
+
+// Start initializes the pipelines
+func (pp *PipelineProvider) NextPipelineChan() chan message.Message {
+	pp.idxMux.Lock()
```
Comment (egueidan) [inline pkg/pipeline/provider.go]: While it's nothing critical, this is typically the case where an atomic would shine.

File: pkg/auditor/auditor.go [hunk 2]
```diff
@@ -245,3 +246,19 @@ func (a *Auditor) unmarshalRegistryV0(b []byte) (map[string]*RegistryEntry, erro
 	}
 	return registry, nil
 }
+
+func (a *Auditor) GetLastCommitedOffsetV0(identifier string) (int64, int) {
```
Comment (egueidan) [inline pkg/auditor/auditor.go]: I feel like this really belongs to `unmarshalRegistryV0`. We should hydrade the v0 registry into a v1 registry with the `file:` prefix.

Comment (tmichelet) [reply pkg/auditor/auditor.go]: way easier indeed

File: pkg/pipeline/provider.go (hunk 1 above)
Comment (tmichelet) [reply pkg/pipeline/provider.go]: +1

**Comment by tmichelet** [comment]
@egueidan updated

**Pull request merged**